boa/src/builtins/array/mod.rs

```diff
@@ -304,6 +305,31 @@ impl Array {
         }
     }
 
+    /// `Array.of(...arguments)`
+    ///
+    /// The Array.of method creates a new Array instance from a variable number of arguments,
+    /// regardless of the number or type of arguments.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-array.of
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/of
+    pub(crate) fn of(this: &Value, args: &[Value], context: &mut Context) -> Result<Value> {
+        let array = match this.as_object() {
+            Some(object) if object.is_constructable() => {
+                object.construct(&[args.len().into()], this.clone(), context)?
+            }
+            _ => Array::array_create(args.len() as u32, None, context)?,
```

Comment:
`Array::array_create` was simplified in #1126 to just return a `Value` since it never returned `Err`. There might be a need for other similar changes, but just running `cargo check` locally should lead down the right path.
```suggestion
            _ => Array::array_create(args.len() as u32, None, context),
```